### PR TITLE
Add unit tests for kubeconfig updates

### LINKED ISSUE #3209: teleport removes existing user definitions in kubeconfg
Have a question? Please use [Our Forum](https://community.gravitational.com)

**What happened**:
When teleport updates kubeconfig, it clears existing `user` settings in kubeconfig.

**What you expected to happen**:
teleport should leave existing user settings in kubeconfig untouched

**How to reproduce it (as minimally and precisely as possible)**:
popular kubeconfig with some use settings and run `tsh login`, you will see existing user settings get updated

```
preferences: {}
users:
- name: arn:aws:eks:x:cluster/x
  user: {}
- name: arn:aws:eks:x:cluster/x
  user: {}
- name: arn:aws:eks:x:cluster/x
  user: {}
- name: arn:aws:eks:x:cluster/x
  user: {}
- name: eks_sfc-dev-k8s-2
  user: {}
```

**Environment**:
- Teleport version (use `teleport version`):
`Teleport v3.2.6 git:v3.2.6-0-g67b4ddfb go1.12.1`
- Tsh version (use `tsh version`):
`Teleport v4.0.2 git:v4.0.2-0-gb7e0e872 go1.12.1`
- OS (e.g. from /etc/os-release):

**Browser environment**

- Browser Version (for UI-related issues):
- Install tools:
- Others:

**Relevant Debug Logs If Applicable**

- tsh --debug
- teleport --debug

### PR DESCRIPTION
**Pull request by awly**
Several tests to confirm correctness of kubeconfig update logic.
Specifically - to make sure existing configuration is not deleted.

`UpdateKubeconfig` was split into two functions because mocking
`*client.TeleportClient` was really difficult.

Fixes #3209

### CODE REVIEW

**Review by webvictim — APPROVED** [review]
LGTM pending a review from one of the others.

**Review by russjones — APPROVED** [review]

**Comment by awly** [comment]
retest this please

**Pull request merged**
